### PR TITLE
Fixes #27072 - remove extra confirm call

### PR DESCRIPTION
**Pull request by upadhyeammit**
This removes the extra confirm_scenario() call in execute_scenario_steps() method . I feel its extra because,

1. The execute_scenario_steps is getting called by two methods, run and run_scenario.
2. The run_scenario is calling confirm_scenario() method.
3. If we see run method, then it calls run_scenario itself, this means we are running confirm_scenario there on condition.

** Only one thing I feel here is when @rescue_scenario is set we are not having confirm_scenario(), so maybe we can call confirm_scenario() when @rescue_scenario is set in execute_scenario_steps() .
** I can see  confirm_scenario() in execute_scenario_steps() added in pr https://github.com/theforeman/foreman_maintain/pull/164
~~~
    def run
      @scenarios.each do |scenario|
        run_scenario(scenario)
        next unless @quit
        if @rescue_scenario
          logger.debug('=== Rescue scenario found. Executing ===')
          execute_scenario_steps(@rescue_scenario, true)
        end
        break
      end
    end

    def run_scenario(scenario, confirm = true)
      return if scenario.steps.empty?
      raise 'The runner is already in quit state' if quit?

      if confirm
        confirm_scenario(scenario)
        return if quit?
      end

      execute_scenario_steps(scenario)
    ensure
      @last_scenario = scenario unless scenario.steps.empty?
      @exit_code = 1 if scenario.failed?
    end
~~~

### CODE REVIEW

**Comment by upadhyeammit** [comment]
@kgaikwad @mbacovsky request to check this, I hooked up pry to understand how code flows to remove that extra call. If I am missing anything here to understand then request to add details.

File: lib/foreman_maintain/runner.rb [hunk 1]
```diff
@@ -92,7 +92,6 @@ def storage
 
     def execute_scenario_steps(scenario, force = false)
       scenario.before_scenarios.flatten.each { |before_scenario| run_scenario(before_scenario) }
-      confirm_scenario(scenario)
```
Comment (kgaikwad) [inline lib/foreman_maintain/runner.rb]: As per you mentioned, it is added for @rescue_scenario. Instead of removing, would it be possible to add condition here which checks if scenario is equal to @rescue_scenario?

Comment (upadhyeammit) [reply lib/foreman_maintain/runner.rb]: Added the change.
Thank you !

**Comment by mbacovsky** [comment]
@upadhyeammit thanks for the patch. I appreciated the detailed explanation!. I have no other concern then what @kgaikwad found (good catch!). 

There are failing tests due to updated Clamp gem that introduces new help formating (https://github.com/mdub/clamp/commit/d18c22d07e71d025e190531fc7c21a43701929c2). We need to update the tests or make them less sensitive for the whitespace, preferably in separate commit or PR.

File: lib/foreman_maintain/runner.rb (hunk 1 above)
Comment (kgaikwad) [reply lib/foreman_maintain/runner.rb]: @upadhyeammit, thanks for adding condition which checks for presence of @rescue_scenario.

Currently, `rescue_scenario` is used by only backup_command.
Not sure but I think, the condition that you added is not enough as `run_scenario` method gets invoked by [backup_command](https://github.com/theforeman/foreman_maintain/blob/master/lib/foreman_maintain/cli/backup_command.rb#L25) where it passes backup_scenario & rescue_scenario as arguments.

There is no any before_scenarios i.e preparation steps for `backup_scenario` argument.
While running this backup_scenario, it will invoke [execute_scenario_steps](https://github.com/theforeman/foreman_maintain/blob/master/lib/foreman_maintain/runner.rb#L51) from `run_scenario`. In this case, the condition that you have added will pass successfully.

So I just want to make sure for `backup_scenario` execution, it should not ask confirmation twice and this confirmation should invoke only when `scenario` inside `execute_scenario_steps` matches with the @rescue_scenario.

**Comment by mbacovsky** [comment]
This PR should fix the failing tests: #275

File: lib/foreman_maintain/runner.rb (hunk 1 above)
Comment (upadhyeammit) [reply lib/foreman_maintain/runner.rb]: I tested this and I dont see confirm call twice for backup and rescue scenarios.

**Comment by upadhyeammit** [comment]
@jameerpathan111 request to have a look. The test failures are of Bash Completion so should you can try out this. Thank You!

**Comment by jameerpathan111** [comment]
> @jameerpathan111 request to have a look. The test failures are of Bash Completion so should you can try out this. Thank You!

sure

**Comment by upadhyeammit** [comment]
Rebased to fix bash test failures.

**Comment by jameerpathan111** [comment]
@upadhyeammit I have tried this PR on Satellite 6.5 and Satellite 6.6 and issue seems to be fixed.
Now pre-migration step in foreman maintain doesn't ask for permission twice.

**Comment by kgaikwad** [comment]
Thanks @upadhyeammit!
